### ISSUE TITLE
examples/encdec cannot work well in GPU

### ISSUE THREAD

**Issue by myohmy** [clab/cnn#26]
Hi all,

Thanks a lot for this great NN tool. I'am happy to see that it provides the GPU version. I have tested the examples using GPU and find that encdec cannot work. The detail is as follows. I further find that the problem occurs in  "loss += as_scalar(cg.forward())"; . If someone knows why, please tell me the reason. Thanks again.

*****************************************************************************************************************
[cnn] using GPU
[cnn] Device Number: 0
[cnn]   Device name: Tesla K80
[cnn]   Memory Clock Rate (KHz): 2505000
[cnn]   Memory Bus Width (bits): 384
[cnn]   Peak Memory Bandwidth (GB/s): 240.48

[cnn]   Memory Free (MB): 1478.89/-805.765

[cnn] Device Number: 1
[cnn]   Device name: Tesla K80
[cnn]   Memory Clock Rate (KHz): 2505000
[cnn]   Memory Bus Width (bits): 384
[cnn]   Peak Memory Bandwidth (GB/s): 240.48

[cnn]   Memory Free (MB): 1575.23/-805.765

[cnn] **USING DEVICE: 1
[cnn] random seed: 1801438780
[cnn] allocating memory: 1536MB
[cnn] memory allocation done.
Reading training data from data/en.40k...
636637 lines, 16006553 tokens, 40003 types
Reading dev data from data/test.en.40k...
919 lines, 25936 tokens
Parameters will be written to: bilm_3_500_500-pid16320.params
**SHUFFLE
before forward computation ...
Segmentation fault (core dumped)
*****************************************************************************************************************

Best,
Jiajun

**Comment by neubig** [comment]
Hi @myohmy , thanks for the report. I've fixed some of the GPU code in cnn so you should no longer be getting segfaults. Unfortunately the example probably still doesn't work though (giving an error that LogSoftmax is not implemented). You can find more information about this problem here, and please feel free to contribute if you're interested:
#28 

I'll close this issue for now, as it's part of the bigger one.

**Comment by myohmy** [comment]
Hi Neubig,

Thank you for the feedback. I know the reason now and I am currently read
the codes in CNN. I will give some reports if I find problems. I will try
my best to do some contribution if I can. Thanks again!

Best,
Jiajun Zhang
